zebrad/src/commands.rs

```diff
@@ -91,14 +91,14 @@ impl ZebradCmd {
 impl Configurable<ZebradConfig> for ZebradCmd {
     /// Location of the configuration file
     fn config_path(&self) -> Option<PathBuf> {
-        let filename = std::env::current_dir().ok().map(|mut dir_path| {
-            dir_path.push(CONFIG_FILE);
-            dir_path
-        });
-
         let if_exists = |f: PathBuf| if f.exists() { Some(f) } else { None };
 
-        filename.and_then(if_exists)
+        dirs::preference_dir()
+            .map(|path| path.join(CONFIG_FILE))
+            .and_then(if_exists)
+            .or_else(|| std::env::current_dir().ok())
+            .map(|path| path.join(CONFIG_FILE))
+            .and_then(if_exists)
```

Comment:
This looks good, combinators are fun!

I think it's ok to use the current directory as a fallback for a config file, because that's much less surprising. And if it's not there, we just use the default.